libgalois/src/PropertyGraph.cpp

```diff
@@ -147,6 +150,60 @@ struct PropertyColumn {
       : field_index(i), array(a) {}
 };
 
+/// MapEntityTypeIDsFromFile takes a file buffer of a node or edge Type set ID file
+/// and extracts the property graph type set ids from it. It is an alternative way
+/// of extracting EntityTypeIDs and extraction from properties will be depreciated in
+/// favor of this method.
+katana::Result<katana::NUMAArray<katana::PropertyGraph::EntityTypeID>>
+MapEntityTypeIDsArray(const tsuba::FileView& file_view) {
+  const auto* data = file_view.ptr<uint32_t>();
+  const int32_t type_ID_array_size = data[0];
+  KATANA_LOG_DEBUG_ASSERT(type_ID_array_size > 0);
+
+  const katana::PropertyGraph::EntityTypeID* type_IDs_array =
+      reinterpret_cast<const katana::PropertyGraph::EntityTypeID*>(data[1]);
+
+  KATANA_LOG_DEBUG_ASSERT(type_IDs_array != nullptr);
+
+  // allocate type IDs array
+  katana::NUMAArray<katana::PropertyGraph::EntityTypeID> type_set_ids;
+  type_set_ids.allocateInterleaved(type_ID_array_size);
+
+  katana::ParallelSTL::copy(
+      &type_IDs_array[0], &type_IDs_array[type_ID_array_size],
+      type_set_ids.begin());
+
+  return katana::Result<katana::NUMAArray<katana::PropertyGraph::EntityTypeID>>(
+      std::move(type_set_ids));
+}
+
+katana::Result<std::unique_ptr<tsuba::FileFrame>>
+WriteEntityTypeIDsArray(
+    const katana::PropertyGraph::EntityTypeID* type_set_ID_array,
+    const uint32_t type_ID_array_size) {
```

Comment:
NUMAArray's size type is `size_t` or you can use `uint64_t`. I prefer `size_t` as that's consistent with STL